flight/Modules/Stabilization/stabilization.c

```diff
@@ -941,13 +951,19 @@ static void SettingsUpdatedCb(UAVObjEvent * ev)
 		// The dT has some jitter iteration to iteration that we don't want to
 		// make thie result unpredictable.  Still, it's nicer to specify the constant
 		// based on a time (in ms) rather than a fixed multiplier.  The error between
-		// update rates on OP (~300 Hz) and CC (~475 Hz) is negligible for this
-		// calculation
+		// update rates on OP (~300 Hz) and CC (~475 Hz) is negligible for this calculation
+		//fakeDt now only used for vbar_decay, for gyro filter now a real, but filtered, dT is used
 		const float fakeDt = 0.0025f;
-		if(settings.GyroTau < 0.0001f)
-			gyro_alpha = 0;   // not trusting this to resolve to 0
+
+		if (settings.GyroCutoff < 0.0001f)
+			gyro_alpha_temp = 0;   // not trusting this to resolve to 0; disable LPF when GyroCutoff = 0
+		else if ( (dT > 0) && (settings.GyroCutoff > (0.45f / dT)) ) { // disable LPF when GyroCutoff > 90% of F_nyquist (= Fs/2 = 1/(2*dT)); using here not the filtered dT, because this is at the start to low
+			gyro_alpha_temp = 0;
+			StabilizationSettingsGyroCutoffSet(&gyro_alpha_temp); // Set GyroCutoff to 0 to signalize in GCS, that it is disabled
+		}
 		else
-			gyro_alpha = expf(-fakeDt  / settings.GyroTau);
+			gyro_alpha_temp = 2.0f * (float)(M_PI) * settings.GyroCutoff;
+
 
 		// Compute time constant for vbar decay term based on a tau
 		vbar_decay = expf(-fakeDt / settings.VbarTau);
```

Comment:
I don't really understand what you mean.

I do there a pre calculation of the LPF coefficient that only changes on UAVObject changes, so not everything has to be calculated every run of the looptime.

Or do you mean i should also change the virtual bar coefficient? As i have absolutely no experience with this flightmode, i don't wanted to touch this, if there is no need.